src/olympia/devhub/tests/test_tasks.py

```diff
@@ -476,6 +500,111 @@ def test_count_listed_signable_addons(self):
         )
 
 
+class TestRunAddonsLinter(ValidatorTestCase):
+
+    def setUp(self):
+        super(TestRunAddonsLinter, self).setUp()
+
+        self.create_switch('webextensions')
+        self.create_switch('addons-linter')
+
+        valid_path = get_addon_file('valid_webextension.xpi')
+        invalid_path = get_addon_file('invalid_webextension_missing_id.xpi')
+
+        self.valid_upload = FileUpload.objects.create(path=valid_path)
+        self.invalid_upload = FileUpload.objects.create(path=invalid_path)
+
+    def get_upload(self, upload):
+        return FileUpload.objects.get(pk=upload.pk)
+
+    @mock.patch('olympia.devhub.tasks.run_addons_linter')
+    def test_run_linter(self, run_linter):
+        run_linter.return_value = '{"errors": 0}'
+
+        assert not self.valid_upload.valid
+
+        tasks.validate(self.valid_upload)
+
+        upload = self.get_upload(self.valid_upload)
+        assert upload.valid, upload.validation
+
+    def test_run_linter_pass(self):
+        assert not self.valid_upload.valid
+
+        tasks.validate(self.valid_upload)
+
+        upload = self.get_upload(self.valid_upload)
+        assert upload.valid, upload.validation
+
+    def test_run_linter_fail(self):
+        tasks.validate(self.invalid_upload)
+        assert not self.get_upload(self.invalid_upload).valid
+
+    def test_run_linter_invalid_file(self):
+        with pytest.raises(ValueError):
+            tasks.run_addons_linter(get_addon_file('empty_file_no_addon'))
+
+    def test_run_linter_path_doesnt_exist(self):
+        with pytest.raises(ValueError) as exc:
+            tasks.run_addons_linter('doesntexist')
+
+        assert exc.value.message == (
+            'Path "doesntexist" is not a file or directory or '
+            'does not exist.\n')
+
+
+class TestValidateFilePath(ValidatorTestCase):
+
```

Comment:
Maybe add a test to see what happens if you don't have the right waffle combination, since you depend on more than one.